### DISCUSSION TITLE
アクセス許可された情報をもとにAPIを呼び出したい

### DISCUSSION THREAD

**Discussion by oyui7272** [lit-iphone-course/lit-iphone-course#35] [Question-original] [answered]
### (1) 実現したいこと
「(位置情報使用の)アクセス許可をする→取得した位置情報を元にAPIを呼び出す→周辺の観光地を表示」の流れを実装したいです。

[現状]
初回立ち上げのときに、アクセス許可のボタンを押し終わると、すでに画面が表示されており(viewDidLoadが呼び出され終わっている?)、APIの呼び出しが終わっています。一度アプリを終了させてから、アプリを再度立ち上げると位置情報を元に周辺の観光地を探すことはできるのですが、一度終了させることが面倒に感じてしまい...🙏

[実現したい流れ]
アクセス許可のボタンを押してから、位置情報を使ってAPIを呼び出すようにしたいです。

### (2) 実現するために自分が試した内容とその結果
アクセス許可を呼び出すところ(SearchViewControllerの110行目)にcallYOLPAPI関数を呼び出すようにする、callYOLPAPI関数をviewDidLoadの一番最後にする(現在地の取得が終わった後)、などを試したのですが、どれもうまくいきませんでした。

### (3) 調べた内容や資料
アクセス許可がどのタイミングで呼ばれているのか、うまく調べられず🙏

### (4) 原因だと思われそうな箇所(ソースコードをコピペして貼る)
関数を呼び出すタイミングが悪いことや、現在地を代入するタイミングが悪いことなどが原因かと考えています。

### (5) GitHubのリンク
https://github.com/yuipi7272/NanoTour.git

**Comment by txxxxc** [comment, marked as answer]
`func locationManager(_ manager: CLLocationManager,didChangeAuthorization status: CLAuthorizationStatus) `
の部分が、アクセスの権限周りに変化があった場合に呼ばれるところなのでこのタイミングでAPIを呼び出してあげればいけると思います！
一回試してみてくださいー！

**Reply by oyui7272** [reply]
遅くなりました！ありがとうございます！

`func locationManager(_ manager: CLLocationManager,didChangeAuthorization status: CLAuthorizationStatus)`の中の
`case .notDetermined`の中に書いてみたのですがうまくいかず...😖

場所の説明が難しいので、slackに画像載せておきますね！

**Reply by txxxxc** [reply]
didChangeAuthorizationの中に書いてみて欲しいですー！

**Reply by txxxxc** [reply]
ごめん！AuthorizedAlwaysの中に書きて欲しいです！（間違えた...🙏）

**Reply by oyui7272** [reply]
できました！！！ありがとうございます😭🙏
(夜に返信してすみません🙇‍♀️)
